### PR TITLE
Update urlparse.py

### PR DESCRIPTION
**Pull request by djallalzoldik**
Hi , team

i was able to convert urlparse.py from urllib that in python3 > https://raw.githubusercontent.com/python/cpython/3.9/Lib/urllib/parse.py

i was able to convert it to be compatible with pytho2
i already test it and it work fine

### CODE REVIEW

**Comment by jeff5** [comment]
Thanks for offering this and for the work behind it.

We haven't previously backported code from Python 3, and I'm wary of starting now. What's the motivation? How much of the URL universe does the existing version not parse? If you have a particular need in your application, there are always ways to patch in an improvement as a local library.

Technicalities if we want to consider this:

- It's in the wrong place. `lib-python` is code we have copied from CPython for re-use and nobody changes it. What you would do is place the same module in the `~/Lib` tree, then the build would use that instead.
- As I understand it, urllib has grown significantly from Python 2 to Python 3. We would want to update the corresponding tests to Python 3. It may work fine for you, but we need to be able to test it every time we build. Again, these belong in the  `~/Lib` tree. This may be a difficult conversion because the test support has changed a lot.
- Your starting version seems to be before the changes that finally landed https://github.com/python/cpython/issues/102153 . I don't know if the current Python 2 library has the same faults, but it seems unwise to expand the range of protocols without fixing a well-publicised CVE.
- If I haven't put you off by now, there's a [contributor agreement](https://www.python.org/psf/contrib/contrib-form/) to sign.

**Comment by jeff5** [comment]
Baseline would be that it passes the existing tests (unless there's a good reason why they're wrong).

File: lib-python/2.7/base58.py
Comment (jeff5) [inline lib-python/2.7/base58.py:1]: This does not appear to have anything to do with the PR.

Comment (djallalzoldik) [reply lib-python/2.7/base58.py:1]: I think all types of encoding are important. for the PR , specially in the new technologies or for the web3

**Pull request closed without merging**